src/plonk/circuit.rs

```diff
@@ -202,58 +314,68 @@ impl<F: Field> Default for ConstraintSystem<F> {
 }
 
 impl<F: Field> ConstraintSystem<F> {
-    /// Add a permutation argument for some advice wires
-    pub fn permutation(&mut self, wires: &[AdviceWire]) -> usize {
+    /// Add a permutation argument for some advice columns
+    pub fn permutation(&mut self, columns: &[Column<Advice>]) -> usize {
         let index = self.permutations.len();
         if self.permutations.is_empty() {
             let at = Rotation(-1);
             let len = self.rotations.len();
             self.rotations.entry(at).or_insert(PointIndex(len));
         }
 
-        for wire in wires {
-            self.query_advice_index(*wire, 0);
+        for column in columns {
+            self.query_advice_index(*column, 0);
         }
-        self.permutations.push(wires.to_vec());
+        self.permutations.push(columns.to_vec());
 
         index
     }
 
-    /// Add a lookup argument for some input wires and table wires.
-    pub fn lookup(&mut self, input_wires: &[InputWire], table_wires: &[TableWire]) -> usize {
+    /// Add a lookup argument for some input columns and table columns.
+    pub fn lookup(
+        &mut self,
+        input_columns: &[Column<Any>],
+        table_columns: &[Column<Any>],
+    ) -> usize {
         let index = self.lookups.len();
         if self.lookups.is_empty() {
             let at = Rotation(-1);
             let len = self.rotations.len();
             self.rotations.entry(at).or_insert(PointIndex(len));
         }
 
-        for input in input_wires {
-            match input {
-                InputWire::Advice(wire) => {
-                    self.query_advice_index(*wire, 0);
+        for input in input_columns {
+            match input.column_type {
+                Any::Advice => {
+                    self.query_advice_index(Column::<Advice>::from(*input), 0);
                 }
-                InputWire::Fixed(wire) => {
-                    self.query_fixed_index(*wire, 0);
+                Any::Fixed => {
+                    self.query_fixed_index(Column::<Fixed>::from(*input), 0);
+                }
+                Any::Aux => {
+                    self.query_aux_index(Column::<Aux>::from(*input), 0);
```

Comment:
Can we avoid this match by pushing it into a general `query_column_index` method?